src/App/Validator/HXL/Metadata/Create.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * Ushahidi Set Validator
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Validator\HXL\Metadata;
+
+use Ushahidi\Core\Entity\ExportJobRepository;
+use Ushahidi\Core\Entity\HXL\HXLLicenseRepository;
+use Ushahidi\Core\Entity\HXL\HXLMetadataRepository;
+use Ushahidi\Core\Entity\UserRepository;
+use Ushahidi\Core\Tool\Validator;
+
+class Create extends Validator
+{
+    protected $repo;
+    protected $user_repo;
+    protected $license_repo;
+    protected $export_job_repo;
+
+    public function __construct(
+        HXLMetadataRepository $repo,
+        UserRepository $user_repo,
+        HXLLicenseRepository $license_repo,
+        ExportJobRepository $export_job_repo
+    ) {
+        $this->repo = $repo;
+        $this->user_repo = $user_repo;
+        $this->license_repo = $license_repo;
+        $this->export_job_repo = $export_job_repo;
+    }
+    /**
+     * @return array|\Ushahidi\Core\Tool\ArrayValidation
+     */
+    protected function getRules()
+    {
+        return array_merge([
+            'private' => [
```

Comment:
@rjmackay  @willdoran code climate complains because of this function being 29 lines. Maybe I could have a function return the rules that are the same as the ones for organisation and dataset_title, but I'm not sure it helps with readability. Tagging you to get your opinion on this

Reply:
I think maybe code climate is just over enthusiastic. 25 lines is really short, and theres barely any real logic in this file, so I don't think it bears splitting it.

Reply:
ah ok, makes sense.